core/src/providers/mistral.rs

```diff
@@ -154,7 +154,11 @@ impl TryFrom<&ChatMessage> for MistralChatMessage {
                 meta_prompt,
                 cm.content.clone().unwrap_or(String::from(""))
             )),
-            name: cm.name.clone(),
+            // Name is only supported for the Function/Tool role.
+            name: match cm.role {
+                ChatMessageRole::Function => cm.name.clone(),
+                _ => None,
```

Comment:
This will end up as `role: null` in the json no ?

Reply:
Nope from my test, the key is removed.

Reply:
It's handled by:
https://github.com/dust-tt/dust/blob/c1af4a7317302fe1773854dde30e6c0185f9fc2f/core/src/providers/mistral.rs#L99-L100

Reply:
We may want to omit the key entirely instead

Reply:
I mean it's the whole point of using serde, right 🙃 ?

Reply:
Indeed, didn't see that (and didn't know serde had  that)

Reply:
(hadn't seen your reply either, github page wasn't refreshed)